[ISSUE #5341, Nuget/Home] Add profiles 44 & 151 to the list of Xamarin supported profiles

DanRigby opened the pull request:
Fixes https://github.com/NuGet/Home/issues/5341

This PR adds PCL Profiles 44 & 151 to the list of Xamarin supported profiles and adds unit tests to cover them.

These profiles have been supported by Xamarin projects since early 2016 but were missing from NuGet's compatibility list.

DanRigby committed:
Add profiles 44 & 151 to Xamarin supported list

* Fixes https://github.com/NuGet/Home/issues/5341

dnfclas commented:
@DanRigby,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

emgarten commented on test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityPCLTests.cs:
```diff
@@ -222,11 +222,17 @@ public void CompatibilityPCL_OptionalUnk(string fw1, string fw2)
         [InlineData("xamarinwatchos10", "portable-net45+win8")]
+        [InlineData("xamarinwatchos10", "portable-net451+netcore451")]
         [InlineData("xamarinmac20", "portable-net45+win8")]
+        [InlineData("xamarinmac20", "portable-net451+netcore451")]
```
Would you add tests for net451+win81  also? It should be the same as net451+netcore451.

emgarten reviewed (changes requested):
Minor comments on tests. Overall this seems fine if 44 and 151 are supported by Xamarin now.

DanRigby committed:
Add Profile 151 tests and adjust TFM naming

- Use win* monikers instead of netcore*.

emgarten reviewed (approved):
LGTM

DanRigby commented:
Updated:
- Used the preferred win* TFM naming
- Added profile 151 tests
- Labeled the profile versions of the tests with comments as done elsewhere in the file

emgarten commented:
@mhutch @mrward  would you take a look also and confirm that these are the right compatibility mappings?

mhutch commented:
Looks good to me!

mrward commented:
Looks good to me.

emgarten commented:
Merged, thanks for the PR!
